app/src/main/java/org/mozilla/fenix/share/ShareController.kt

```diff
@@ -64,7 +67,16 @@ class DefaultShareController(
             flags = FLAG_ACTIVITY_NEW_TASK
             setClassName(app.packageName, app.activityName)
         }
-        fragment.startActivity(intent)
+
+        try {
+            fragment.startActivity(intent)
+        } catch (e: SecurityException) {
+            context.getRootView()?.let {
+                FenixSnackbar.make(it, Snackbar.LENGTH_LONG)
+                    .setText(it.context.getString(R.string.share_error_snackbar)
+                ).show()
```

Comment:
Nit: Just a minor format issue, but consider putting the `)` before `.show()` at the end of the previous line?